test/suite/test_cc12.py

```diff
@@ -0,0 +1,100 @@
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wttest
+from wiredtiger import stat
+from test_cc01 import test_cc_base
+from helper import WiredTigerStat
+from wtscenario import make_scenarios
+import random
+import threading
+import os
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+# test_cc12.py
+# Verify checkpoint cleanup cleans up logged tables when configured in aggressive mode.
+@wttest.skip_for_hook("tiered", "Checkpoint cleanup does not support tiered tables")
+class test_cc12(test_cc_base):
+    conn_config = 'statistics=(all),statistics_log=(json,wait=1,on_close=true),log=(enabled=true)'
+
+    checkpoint_cleanup_methods = [
+        ('cc_method_reclaim_space', dict()),
+    ]
+
+    scenarios = make_scenarios(checkpoint_cleanup_methods)
+
+    lock = threading.Lock()
+    g_ts = 1
+
+    def random_populate(self, uri, seed, ts):
+        session = self.conn.open_session()
+        for r in range(1000000):
+            session.begin_transaction()
+            cur = session.open_cursor(uri)
+            for sr in range(1000):
+                key = random.randint(0, 10000)
+                cur[key] = f"upd_{r}_{sr}_{seed}"
+            cur.close()
+            with self.lock:
+                ts = self.g_ts
+                self.g_ts += 1
+            session.commit_transaction(f'commit_timestamp=' + self.timestamp_str(ts))
+
+    def long_transaction_round(self, uri, ts):
+        self.session.begin_transaction()#'read_timestamp=' + self.timestamp_str(1))
+        long_cur = self.session.open_cursor(uri)
+        long_cur[-1] = "halted"
+        long_cur.close()
+        with ThreadPoolExecutor(max_workers=20) as executor:
+            for seed in range(10000):
+                # self.random_populate(uri, seed, ts)
+                executor.submit(self.random_populate, uri, seed, ts)
+        self.session.commit_transaction()
+        self.session.checkpoint()
+
+
+    def test_cc12(self):
+        # Increase the likelihood of having internal pages since they are targeted by checkpoint
+        # cleanup.
+        create_params = 'key_format=i,value_format=S,allocation_size=512,internal_page_max=512,leaf_page_max=512'
+        uri = 'table:cc12'
+
+        self.session.create(uri, create_params)
+        
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(1))
+        history_size = []
```

Comment:
There is stat also that we track for the history store table size, if you want to use that.